ui/src/App.tsx

```diff
@@ -151,13 +151,25 @@ export default function App() {
       const existFlowIds = listWorkflows().map((flow) => flow.id);
       const { fileList, folderList } = await scanLocalNewFiles(
         myWorkflowsDir!,
-        existFlowIds
+        existFlowIds,
       );
       await syncNewFlowOfLocalDisk(fileList, folderList);
     }
   };
 
   useEffect(() => {
+    /**
+     * because we have turned on strict mode, useEffect will be executed twice in strict mode in the development environment.
+     * and graphAppSetup contains DB related operations, repeated execution will bring some bad results.
+     * so in development environment mode, the first execution is skipped.
+     */
+    if (
+      process.env.NODE_ENV === "development" &&
+      developmentEnvLoadFirst.current
+    ) {
+      developmentEnvLoadFirst.current = false;
+      return;
+    }
     graphAppSetup();
     setLoadChild(true);
     setInterval(() => {
```

Comment:
i forgot to add clean up for the timer, could you please help add a cleanup timer when demount?

Reply:
has been added, and two useEffects have been merged into one.